pkg/controllermanager/controller/seed/seed_lifecycle_reconcile.go

```diff
@@ -127,6 +129,25 @@ func (c *livecycleReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 		}
 	}
 
+	// If the gardenlet's client certificate is expired and the seed belongs to a `ManagedSeed` then we reconcile it in
+	// order to re-bootstrap the gardenlet.
+	if seed.Status.ClientCertificateExpirationTimestamp != nil && seed.Status.ClientCertificateExpirationTimestamp.UTC().Before(time.Now().UTC()) {
+		managedSeed := &seedmanagementv1alpha1.ManagedSeed{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      seed.Name,
+				Namespace: v1beta1constants.GardenNamespace,
+			},
+		}
+		if err := c.gardenClient.Client().Get(ctx, client.ObjectKeyFromObject(managedSeed), managedSeed); err != nil && !apierrors.IsNotFound(err) {
+			return reconcileResult(err)
+		} else if err == nil {
+			metav1.SetMetaDataAnnotation(&managedSeed.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile)
```

Comment:
Perhaps you could add a log message before this line (at least with `Debug` level)?